store/etcd/etcd.go

```diff
@@ -470,6 +567,32 @@ func (s *Etcd) List(directory string) ([]*store.KVPair, error) {
 	return kv, nil
 }
 
+// ListExt lists child nodes of a given directory, etcd native extension.
+func (s *Etcd) ListExt(directory string) (*store.KVPairExt, error) {
```

Comment:
there is difference between watching a tree an getting a list back:
* for watching a tree, either you always send a list (i.e whole tree) back when there is a change like in the original watchtree, or give an exact node in the tree back but they *both* still watch over the whole tree.
* for list, you expect a list back, you don't expect a single pair back, since the command itself is for listing, so you can't match that with how the watchtree/ext works.